admin-ui/plugins/auth-server/helper/validations.js

```diff
@@ -0,0 +1,25 @@
+// LDAP validation schema using Yup
+import * as Yup from 'yup'
+import { STRINGS } from './constants'
+
+export function getLdapValidationSchema() {
+  const e = STRINGS.authn.ldap.errors
+  return Yup.object().shape({
+    configId: Yup.string().min(1, e.acr).required(e.acr),
+    bindDN: Yup.string().min(1, e.bind_dn).required(e.bind_dn),
+    bindPassword: Yup.string().min(1, e.bind_password).required(e.bind_password),
+    servers: Yup.array()
+      .of(Yup.string().min(1, e.remote_ldap_server).required(e.remote_ldap_server))
+      .min(1, e.remote_ldap_server)
+      .required(e.remote_ldap_server),
+    baseDNs: Yup.array()
+      .of(Yup.string().min(1, e.base_dns).required(e.base_dns))
+      .min(1, e.base_dns)
+      .required(e.base_dns),
+    maxConnections: Yup.number().typeError(e.max_connections_type).required(e.max_connections),
```

Comment:
`Failing to validate that a numeric input is a positive number is bad because it violates the principle of robust user input validation by only checking for the data type and presence of the value, which allows invalid data like zero or negative numbers to pass client-side checks.`